pkg/reconciler/workload/apiexport/workload_apiexport_reconcile.go

```diff
@@ -106,25 +87,26 @@ func (r *schemaReconciler) reconcile(ctx context.Context, export *apisv1alpha1.A
 		}
 		schemaName := fmt.Sprintf("%s.%s", resource, group)
 
-		// APIResourceSchemas already in root:compute should be skipped.
-		if shouldSkip && rootComputeResourceSchema.Has(schemaName) {
-			logger.V(4).Info("Skipping resource that's already in root:compute", "resource", schemaName)
-			continue
-		}
-
-		expectedResourceGroups.Insert(schemaName)
 		resourcesByResourceGroup[schemaName] = r
 	}
 
 	// reconcile schemas in export
-	upToDate := sets.NewString()
-	schemasByResourceGroup := map[string]*apisv1alpha1.APIResourceSchema{}
+	// we check all schemas reference in the apiexport. if it is missing, we should create the schema
+	// if it is outdated, we should create the schema and delete the outdated schema.
+	upToDateResourceGroup := sets.NewString()
+	expectedResourceGroups := sets.NewString()
+
+	// schemaNamesByResourceGroup records the upToData APIResourceSchemaName for the resourceGroup, and the
+	// APIExport will be updated accordingly
+	schemaNamesByResourceGroup := map[string]string{}
 	for _, schemaName := range export.Spec.LatestResourceSchemas {
 		_, resource, group, ok := split3(schemaName, ".")
```

Comment:
Why do we have an entirely different splitting mechanism here? Can we collapse these? And, can we use some more strongly-formatted data? Does validation on APIResourceSchema require that the name be `revision.resource.group`?

Reply:
yes, it requires the schema name to be revision.resource.group by admission.